index.js

```diff
@@ -1,168 +1,143 @@
+const store = new Corestore(argv.storage || './corestore')
+
+let swarm = null
+let tracker = null
+let stdout = ''
 
-start().catch(err => {
+main().catch(err => {
   console.error(err)
   process.exit(1)
 })
 
-async function start () {
-  const secretKey = argv['secret-key']
-  const store = new Corestore(argv.storage || './corestore')
-  const port = argv.port
-
-  const dht = new DHT({ port })
-  const swarm = new Hyperswarm({
+async function main () {
+  swarm = new Hyperswarm({
     seed: secretKey ? HypercoreId.decode(secretKey) : undefined,
     keyPair: secretKey ? undefined : await store.createKeyPair('simple-seeder-swarm'),
-    dht
+    dht: new DHT({ port: argv.port })
   })
-
-  tracking.swarm = swarm
-
-  const cores = [].concat(argv.core || argv.key || [])
-  const bees = [].concat(argv.bee || [])
-  const drives = [].concat(argv.drive || [])
-  const seeders = [].concat(argv.seeder || [])
-
-  if (argv.file) {
-    const seeds = await fs.promises.readFile(argv.file)
-    for (const [type, key] of configs.parse(seeds, { split: ' ', length: 2 })) {
-      if (type === 'core' || type === 'key') cores.push(key)
-      else if (type === 'bee') bees.push(key)
-      else if (type === 'drive') drives.push(key)
-      else if (type === 'seeder') seeders.push(key)
-      else throw new Error('Invalid seed type: ' + type)
-    }
-  }
-
   swarm.on('connection', onsocket)
   swarm.listen()
+  goodbye(() => swarm.destroy())
 
-  for (const key of cores) await downloadCore(key)
-  for (const key of bees) await downloadBee(key)
-  for (const key of drives) await downloadDrive(key)
+  if (argv.menu) {
+    const menu = require('./menu.js')
+    await menu(argv.menu, { store, swarm })
+    console.log('Closing')
+    goodbye.exit()
+    return
+  }
 
-  for (const key of seeders) {
-    const publicKey = HypercoreId.decode(key)
-    const id = HypercoreId.encode(publicKey)
-    const keyPair = await store.createKeyPair('simple-seeder-swarm@' + id)
-    const sw = new Seeders(publicKey, { dht: swarm.dht, keyPair })
+  const seeds = await load()
 
-    tracking.seeders.push(sw)
+  tracker = new SimpleSeeder(store, swarm, { backup: argv.backup })
+  goodbye(() => tracker.destroy())
 
-    sw.join()
-    sw.on('connection', onsocket)
-    sw.on('update', function (record) {
-      tracking.notifs[id] = record
-    })
+  for (const { key, type } of seeds) {
+    if (type === 'seeders' && seeds.find(s => s.type === 'drive' && s.key === key)) continue
+    const seeders = !!(type === 'drive' && seeds.find(s => s.type === 'seeders' && s.key === key))
 
-    goodbye(() => sw.destroy())
+    await tracker.add(key, { type, seeders })
+  }
 
-    if (!drives.includes(key)) {
-      await downloadDrive(key, false)
-    }
+  const intervalId = setInterval(ui, argv.i || 5000)
+  goodbye(() => clearInterval(intervalId)) // Explicit cleanup
+  ui()
+
+  const lists = tracker.filter(r => r.type === 'list')
+  if (lists[0] && !argv['dry-run']) {
```

Comment:
More of a detail, and no need to change, but I prefer to extract the relevant argv's at the start of my code and to then pass them around as variables instead of accessing argv throughout the code, creates more of a separation between the input and the processing code